packages/hop-node/src/db/TransferRootsDb.ts

```diff
@@ -180,58 +276,57 @@ class TransferRootsDb extends TimestampedKeysDb<TransferRoot> {
     return normalizeDbItem(item)
   }
 
-  async getByTransferRootHash (
-    transferRootHash: string
+  async getByTransferRootId (
+    transferRootId: string
   ): Promise<TransferRoot> {
-    const item: TransferRoot = await this.getById(transferRootHash)
+    await this.tilReady()
+    const item: TransferRoot = await this.getById(transferRootId)
     return this.normalizeItem(item)
   }
 
-  async getByTransferRootId (transferRootId: string): Promise<TransferRoot | null | undefined> {
-    const transferRootHashes = await this.getTransferRootHashes()
-    const filtered = (
-      await Promise.all(
-        transferRootHashes.map(async (transferRootHash: string) => {
-          const item = await this.getByTransferRootHash(transferRootHash)
-          if (item.transferRootId === transferRootId) {
-            return item
-          }
-        })
-      )
-    ).filter(this.filterExisty)
-    return filtered?.[0]
+  async getByTransferRootHash (
+    transferRootHash: string
+  ): Promise<TransferRoot | null> {
+    await this.tilReady()
+    let item = await this.subDbRootHashes.getById(transferRootHash)
+    if (!item?.transferRootId) {
+      return null
+    }
+    item = await this.getById(item.transferRootId)
+    return this.normalizeItem(item)
   }
 
   filterTimestampedKeyValues = (x: any) => {
-    return x?.value?.transferRootHash
+    return x?.value?.transferRootId
   }
 
-  async getTransferRootHashes (dateFilter?: TransferRootsDateFilter): Promise<string[]> {
+  async getTransferRootIds (dateFilter?: TransferRootsDateFilter): Promise<string[]> {
+    await this.tilReady()
     // return only transfer-root keys that are within specified range (filter by timestamped keys)
+    const filter: KeyFilter = {
```

Comment:
Is there a reason why these filters are defined here now instead of only within the `if` statements if the `dateFilter` exists? If there is a good reason, why did this not exist before or why does it not exist in the Transfers DB?

Reply:
The reason is to only use the timestamped keys db for returning the ids, with or without filter timestamps, instead of using the base db or timestamped keys db depending whether there's a filter or not. It simplifies the function. The same logic has been applied to the TransfersDb now in https://github.com/hop-protocol/hop/pull/318/commits/ee413eece6af209e454ded15ba8fa64822376447

Reply:
Oh nice that is an awesome update! I really like that.

Only suggestion now is to remove the newly-unused `filterOutTimestampedKeys()`.

Reply:
Removed in https://github.com/hop-protocol/hop/pull/318/commits/75db07adbd7005fadbda27102c80d24358b9789b